staging/src/k8s.io/client-go/tools/record/event_test.go

```diff
@@ -101,6 +102,29 @@ func OnPatchFactory(testCache map[string]*v1.Event, patchEvent chan<- *v1.Event)
 	}
 }
 
+func TestNonRacyShutdown(t *testing.T) {
+	// Attempt to simulate previously racy conditions, and ensure that no race
```

Comment:
the test by itself does not assert no race condition. it would rely on the test being run with race detector enabled.

Reply:
IIRC, that's not true -- the "race" is pretty easily detected as a write to a closed channel, which works even w/o the race detector (it just panics)